### PR TITLE
UAR-329: Implement continue with saved filing page journey

### PR DESCRIPTION
**Pull request by mmcpolin-ch**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-329

### Change description
Add continue with saved filing page to update Journey

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/middleware/authentication.middleware.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const authentication = (req: Request, res: Response, next: NextFunction):
       let returnUrl = SOLD_LAND_FILTER_URL;
 
       if (req.path.startsWith(UPDATE_LANDING_URL)) {
-        returnUrl = SECURE_UPDATE_FILTER_URL;
+        returnUrl = UPDATE_CONTINUE_WITH_SAVED_FILING_URL;
```
Comment (AndreKramer7654) [inline src/middleware/authentication.middleware.ts:25]: This only if save&resume is enabled. The page is only to be shown if the feature is on.

**Comment by AndreKramer7654** [comment]
Should this page just be a normal part of the journey (when update save & resume flag is on)? I don't see why it needs to be a follow on from authentication - you could already be authenticated when you go to the update-an-overseas-entity landing page.

File: views/update/continue-with-saved-filing.html [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+{% endblock %}
+
+{% block content %}
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+
+      {% include "includes/list/errors.html" %}
+
+      <form method="post">
+
```
Comment (sjahir) [inline views/update/continue-with-saved-filing.html:16]: remove empty line

File: views/update/continue-with-saved-filing.html [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+              },
+              attributes: {
+                "data-event-id": "no-starting-new-filing"
+              }
+            }
+          ]
+        }) }}
+
+        {% include "includes/continue-button.html" %}
+
```
Comment (sjahir) [inline views/update/continue-with-saved-filing.html:57]: remove empty line

File: src/middleware/service.availability.middleware.ts [hunk 4]
```diff
@@ -35,6 +37,14 @@ export const serviceAvailabilityMiddleware = (req: Request, res: Response, next:
     return res.render(SERVICE_OFFLINE_PAGE);
   }
 
+  if (
+    (req.path === UPDATE_CONTINUE_WITH_SAVED_FILING_URL || req.path.endsWith(`/${RESUME}`))
+    &&
+    !isActiveFeature(FEATURE_FLAG_ENABLE_UPDATE_SAVE_AND_RESUME)) {
```
Comment (AndreKramer7654) [inline src/middleware/service.availability.middleware.ts:43]: I don't understand the intention here. The UPDATE_CONTINUE_WITH_SAVED_FILING_URL  page should only be re-directed from the start of the journey. The landing URL is different when save&resume is enabled I believe and the only way to get to it is from "/update-an-overseas-entity/" path? If this is only an extra check to make sure that is the case then fine.

File: src/routes/index.ts [hunk 5]
```diff
@@ -487,4 +488,9 @@ router.route(config.UPDATE_FILING_DATE_URL)
   .get(updateFilingDate.get)
   .post(updateFilingDate.post);
 
+router.route(config.UPDATE_CONTINUE_WITH_SAVED_FILING_URL)
+  .all(authentication)
+  .get(updateContinueSavedFiling.get)
+  .post(...validator.updateContinueSavedFiling, checkValidations, updateContinueSavedFiling.post);
```
Comment (AndreKramer7654) [inline src/routes/index.ts:494]: do we need both validator.updateContinueSavedFiling and checkValidations?

**Comment by AndreKramer7654** [comment]
Do we need a change to the web app env config to set the landing page to be this new page? Only if save& resume is enabled? Our landing controller should then re-direct to it (no code change needed unless we re-direct there based on the feature flag)? If you go to "/update-an-overseas-entity"?

**Review by sjahir — APPROVED** [review]

File: src/middleware/service.availability.middleware.ts (hunk 4 above)
Comment (AndreKramer7654) [reply src/middleware/service.availability.middleware.ts:43]: Probably overly restrictive but as done by registration

**Review by AndreKramer7654 — APPROVED** [review]

**Review by jkayyala — APPROVED** [review]

**Review by bantao-ch — APPROVED** [review]
@mmcpolin-ch -- please resolve conflicts on the branch

**Review by markpit — APPROVED** [review]

**Pull request merged**